docker-compose-prod.yml

```diff
@@ -0,0 +1,63 @@
+# Version 3, most current, and recommended compose file version
+# https://docs.docker.com/compose/compose-file/compose-versioning/
+version: '3'
+
+services:
+  # First service
+  telescope:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    env_file: .env
+    environment:
+      # a place-holder for the variable passed by CLI with a default value
+      # the passed or default value is a command to run `telescope`
+      - script=${script:-start}
+      # override the .env variable for `REDIS_URL`
+      - REDIS_URL=redis://redis:${REDIS_PORT}
+    depends_on:
+      - redis
+    ports:
+      - '$PORT:$PORT'
+    restart: always
+
+  # Second service
+  redis:
+    image: redis:latest
+    ports:
+      - '$REDIS_PORT:$REDIS_PORT'
+
+  # Third service
+  # SSO Identity Provider test service, https://simplesamlphp.org
+  # Access to the authentication page via http://localhost:8080/simplesaml or https://localhost:8443/simplesaml
+  ssoipd:
+    environment:
+      - SIMPLESAMLPHP_SP_ENTITY_ID=${SAML2_CLIENT_ID:-saml-poc}
+      - SIMPLESAMLPHP_SP_ASSERTION_CONSUMER_SERVICE=${SAML2_REDIRECT_URI}
+    # image owner's blog post https://medium.com/disney-streaming/setup-a-single-sign-on-saml-test-environment-with-docker-and-nodejs-c53fc1a984c9
+    image: kristophjunge/test-saml-idp
+    ports:
+      # These ports + port 80 are pre-defined in the latter image
+      # http port
+      - '8080:8080'
+      # https port
+      - '8443:8443'
+    restart: always
+  nginx:
+    image: nginx:latest
+    container_name: production_nginx
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+      - ../certbot/conf:/etc/letsencrypt
+      - ../certbot/www:/var/www/certbot
+    ports:
+      - 80:80
+      - 443:443
+    restart: always
+    command: '/bin/sh -c ''while :; do sleep 6h & wait $${!}; nginx -s reload; done & nginx -g "daemon off;"'''
```

Comment:
Can you add a comment above this line explaining why you're doing this 6h dance?